csharp/CSharpParser.g4

```diff
@@ -182,16 +214,26 @@ primary_expression_start
 	             | rank_specifier+ array_initializer)
 	      | anonymous_object_initializer
 	      | rank_specifier array_initializer)                       #objectCreationExpression
+	| OPEN_PARENS argument ( ',' argument )+ CLOSE_PARENS           #tupleExpression
 	| TYPEOF OPEN_PARENS (unbound_type_name | type_ | VOID) CLOSE_PARENS   #typeofExpression
 	| CHECKED OPEN_PARENS expression CLOSE_PARENS                   #checkedExpression
 	| UNCHECKED OPEN_PARENS expression CLOSE_PARENS                 #uncheckedExpression
-	| DEFAULT OPEN_PARENS type_ CLOSE_PARENS                         #defaultValueExpression
+	| DEFAULT (OPEN_PARENS type_ CLOSE_PARENS)?                     #defaultValueExpression
 	| ASYNC? DELEGATE (OPEN_PARENS explicit_anonymous_function_parameter_list? CLOSE_PARENS)? block #anonymousMethodExpression
 	| SIZEOF OPEN_PARENS type_ CLOSE_PARENS                          #sizeofExpression
 	// C# 6: https://msdn.microsoft.com/en-us/library/dn986596.aspx
 	| NAMEOF OPEN_PARENS (identifier '.')* identifier CLOSE_PARENS  #nameofExpression
 	;
 
+throw_expression
+	: THROW expression
+	;
+
+throwable_expression
+	: expression
+	| throw_expression
+	;
+
```

Comment:
Please reorder because `throw_expression` is used below `throwable_expression`:

```g4
throwable_expression
	: expression
	| throw_expression
	;

throw_expression
	: THROW expression
	;
```